### PR TITLE
Wait for snapshot to be ready before restoring (for azure)

### PR DESCRIPTION
**Pull request by sseago**

### CODE REVIEW

**Review by eriknelson — APPROVED** [review]
I've tested this but I'm unfamiliar with this code, holding on a merge for someone who knows better.

File: pkg/cloudprovider/azure/volume_snapshotter.go [hunk 1]
```diff
@@ -185,6 +203,58 @@ func (b *VolumeSnapshotter) CreateVolumeFromSnapshot(snapshotID, volumeType, vol
...
+			return false, nil
+		}
+		if *snapshot.DiskProperties.ProvisioningState == "Succeeded" {
+			return true, nil
+		}
+		if *snapshot.DiskProperties.ProvisioningState == "Failed" {
+			logger.Debug("snapshot has 'Failed' ProvisioningState")
+			return true, errors.Errorf("Snapshot has 'Failed' ProvisioningState")
+		}
+		if *snapshot.DiskProperties.ProvisioningState == "Failed" {
```
Comment (pranavgaikwad) [inline pkg/cloudprovider/azure/volume_snapshotter.go]: @sseago Is this intended? should it be 'Deleting'?

Comment (sseago) [reply pkg/cloudprovider/azure/volume_snapshotter.go]: @pranavgaikwad fixed

**Review by pranavgaikwad — APPROVED** [review]

**Comment by pranavgaikwad** [comment]
LGTM

**Pull request merged**